apps/trading/lib/hooks/use-current-programs.ts

```diff
@@ -0,0 +1,289 @@
+  infrastructureFee: number;
+  liquidityFee: number;
+  makerFee: number;
+  treasuryFee: number;
+};
+
+type RawFactors = Omit<
+  | ReferralProgramFieldsFragment['benefitTiers'][0]['referralRewardFactors']
+  | ReferralProgramFieldsFragment['benefitTiers'][0]['referralDiscountFactors']
+  | VolumeDiscountFieldsFragment['benefitTiers'][0]['volumeDiscountFactors'],
+  '__typename'
+>;
+
+export type Factors = {
+  infrastructureFactor: number;
+  makerFactor: number;
+  liquidityFactor: number;
+};
+
+export type ReferralBenefitTier = {
+  tier: number;
+  discountFactors: Factors;
+  /** calculated min discount factor for the tier */
+  discountFactor: BigNumber;
+  rewardFactors: Factors;
+  /** calculated min reward factor for the tier */
+  rewardFactor: BigNumber;
+  minimumRunningNotionalTakerVolume: number;
+  epochs: number;
+};
+
+export type ReferralStakingTier = {
+  tier: number;
+  minimumStakedTokens: string;
+  referralRewardMultiplier: string;
+};
+
+export type VolumeDiscountBenefitTier = {
+  tier: number;
+  discountFactors: Factors;
+  /** calculated min discount factor for the tier */
+  discountFactor: BigNumber;
+  minimumRunningNotionalTakerVolume: number;
+};
+
+export type ProgramsData = {
+  referralProgram:
+    | {
+        details: Omit<
+          ReferralProgramFieldsFragment,
+          'benefitTiers' | 'stakingTiers'
+        >;
+        benefitTiers: ReferralBenefitTier[];
+        stakingTiers: ReferralStakingTier[];
+      }
+    | undefined;
+  volumeDiscountProgram:
+    | {
+        details: Omit<VolumeDiscountFieldsFragment, 'benefitTiers'>;
+        benefitTiers: VolumeDiscountBenefitTier[];
+      }
+    | undefined;
+  loading: boolean;
+  error?: ApolloError;
+};
+
+export const useCurrentPrograms = (): ProgramsData => {
+  const { data, loading, error } = useCurrentProgramsQuery({
+    errorPolicy: 'ignore',
+    fetchPolicy: 'cache-and-network',
+  });
+
+  const referralProgramData = data?.currentReferralProgram;
+  const volumeDiscountProgramData = data?.currentVolumeDiscountProgram;
+
+  const fees = getFees(data);
+
+  let referralProgram = undefined;
+  if (referralProgramData) {
+    const benefitTiers = prepareReferralBenefitTiers(referralProgramData, fees);
+    const stakingTiers = prepareReferralStakingTiers(referralProgramData);
+    const details = omit(referralProgramData, 'benefitTiers', 'stakingTiers');
+
+    referralProgram = {
+      benefitTiers,
+      stakingTiers,
+      details,
+    };
+  }
+
+  let volumeDiscountProgram = undefined;
+  if (volumeDiscountProgramData) {
+    const benefitTiers = prepareVolumeDiscountBenefitTiers(
+      volumeDiscountProgramData,
+      fees
+    );
+    const details = omit(volumeDiscountProgramData, 'benefitTiers');
+
+    volumeDiscountProgram = {
+      benefitTiers,
+      details,
+    };
+  }
+
+  return {
+    referralProgram,
+    volumeDiscountProgram,
+    loading,
+    error,
+  };
+};
+
+function prepareReferralBenefitTiers(
```

Comment:
Lots of data manipulation and calculation in this file. It would benefit from a unit test